rest-api/resources/src/main/java/org/eclipse/kapua/app/api/resources/v1/resources/DataMessages.java

```diff
@@ -88,7 +93,8 @@
             @QueryParam("metricMin") String metricMinValue, //
             @QueryParam("metricMax") String metricMaxValue, //
             @QueryParam("offset") @DefaultValue("0") int offset,//
-            @QueryParam("limit") @DefaultValue("50") int limit) throws KapuaException {
+            @QueryParam("limit") @DefaultValue("50") int limit,
+            @QueryParam("order") @DefaultValue("desc") String order) throws KapuaException {
```

Comment:
A few of things:

- Instead of a `String`, you could use the `SortDirection` type itself so it's automatically parsed, and invalid values are rejected earlier. Also, to have it case-insensitive, please implement the `fromString()` method in `SortDirection`. You can draw inspiration from #2891
- Also, for coherence with #2891, I would rather name the parameter `sortDir`
- This is just a stylistic preference, but I'd like to keep `offset` and `limit` to end of the signature, so could you please move `sortDir` just below `metricMax`?